src/index.ts

```diff
@@ -189,56 +221,12 @@ function fetchAndLaunchMetals(context: ExtensionContext, javaHome: string) {
     });
 }
 
-function launchMetals(
+function registerCommands(
   context: ExtensionContext,
-  javaPath: string,
-  metalsClasspath: string,
-  serverProperties: string[],
-  javaOptions: string[],
-  env: { COURSIER_REPOSITORIES?: string }
+  client: LanguageClient,
+  features: MetalsFeatures,
+  treeViewsManager: TreeViewsManager
 ) {
-  const baseProperties = [
-    `-Dmetals.client=coc-metals`,
-    `-Dmetals.doctor-format=json`,
-    `-Xss4m`,
-    `-Xms100m`
-  ];
-  const mainArgs = ["-classpath", metalsClasspath, "scala.meta.metals.Main"];
-  // let user properties override base properties
-  const launchArgs = baseProperties
-    .concat(javaOptions)
-    .concat(serverProperties)
-    .concat(mainArgs);
-
-  const serverOptions: ServerOptions = {
-    run: { command: javaPath, args: launchArgs, options: { env } },
-    debug: { command: javaPath, args: launchArgs, options: { env } }
-  };
-
-  const clientOptions: LanguageClientOptions = {
-    documentSelector: [{ scheme: "file", language: "scala" }],
-    synchronize: {
-      configurationSection: "metals"
-    },
-    revealOutputChannelOn: RevealOutputChannelOn.Never
-  };
-
-  // TODO fix this any
-  // I know this any here is a bit gross, but I can't
-  // get any of the sendRequests to work or type check
-  // correctly without it. I notice that this was also
-  // done in some of the other coc-extensions, which
-  // game me the idea to try it this way, and it works
-  const client: any = new LanguageClient(
```

Comment:
Can you explain your reasoning for moving the client creation to up above instead of leaving it here?